docs/design/coreclr/jit/investigate-stress.md

```diff
@@ -0,0 +1,19 @@
+# Investigating JIT and GC stress
+
+There are two stressing related features for the JIT and JIT generated GC info &ndash; JIT Stress and GC Stress. These features provide a way during development to discover edge cases and more "real world" scenarios without having to develop complex applications.
+
+## JIT Stress (Debug builds only)
+
+Enabling JIT Stress can be done in several ways. Setting `DOTNET_JitStress` to a non-zero integer value that will generate varying levels of JIT optimizations based on a hash of the method's name or set to a value of two (for example, `DOTNET_JitStress=2`) that will apply all optimizations. Another way to enable JIT Stress is by setting `DOTNET_JitStressModeNamesOnly=1` and then requesting the stress modes, space delimited, in the `DOTNET_JitStressModeNames` variable (for example, `DOTNET_JitStressModeNames=STRESS_USE_CMOV STRESS_64RSLT_MUL STRESS_LCL_FLDS`).
+
```

Comment:
Maybe add a link to the list of mode types.

Reply:
@BruceForstall Are these modes officially documented somewhere? I found a list in `compiler.h` but I'm not sure how to express that in doc. I guess I could link to the file and then search for `STRESS_MODES`?

Reply:
They're not documented. The names come from the `STRESS_MODES` define:
https://github.com/dotnet/runtime/blob/81f884d1a5f4bbd562908ed8aa84ba9d6fc96e65/src/coreclr/jit/compiler.h#L9621-L9672

In a JitDump, you'll see a line like:
```
*** JitStress: STRESS_NULL_OBJECT_CHECK ***
```
indicating that one has kicked in.

Reply:
@BruceForstall Agreed. I was trying to follow up on the initial statement above. I figured there was one based on the recommendation. Do you think I should just going with my proposal?

> I guess I could link to the file and then search for STRESS_MODES?